proposals/native-integers.md

```diff
@@ -0,0 +1,342 @@
+| `+` | `nuint operator +(nuint left, nuint right)` | `add` / `add.ovf.un` |
+| `-` | `nint operator -(nint left, nint right)` | `sub` / `sub.ovf` |
+| `-` | `nuint operator -(nuint left, nuint right)` | `sub` / `sub.ovf.un` |
+| `*` | `nint operator *(nint left, nint right)` | `mul` / `mul.ovf` |
+| `*` | `nuint operator *(nuint left, nuint right)` | `mul` / `mul.ovf.un` |
+| `/` | `nint operator /(nint left, nint right)` | `div` |
+| `/` | `nuint operator /(nuint left, nuint right)` | `div.un` |
+| `%` | `nint operator %(nint left, nint right)` | `rem` |
+| `%` | `nuint operator %(nuint left, nuint right)` | `rem.un` |
+| `==` | `bool operator ==(nint left, nint right)` | `beq` / `ceq` |
+| `==` | `bool operator ==(nuint left, nuint right)` | `beq` / `ceq` |
+| `!=` | `bool operator !=(nint left, nint right)` | `bne` |
+| `!=` | `bool operator !=(nuint left, nuint right)` | `bne` |
+| `<` | `bool operator <(nint left, nint right)` | `blt` / `clt` |
+| `<` | `bool operator <(nuint left, nuint right)` | `blt.un` / `clt.un` |
+| `<=` | `bool operator <=(nint left, nint right)` | `ble` |
+| `<=` | `bool operator <=(nuint left, nuint right)` | `ble.un` |
+| `>` | `bool operator >(nint left, nint right)` | `bgt` / `cgt` |
+| `>` | `bool operator >(nuint left, nuint right)` | `bgt.un` / `cgt.un` |
+| `>=` | `bool operator >=(nint left, nint right)` | `bge` |
+| `>=` | `bool operator >=(nuint left, nuint right)` | `bge.un` |
+| `&` | `nint operator &(nint left, nint right)` | `and` |
+| `&` | `nuint operator &(nuint left, nuint right)` | `and` |
+| <code>&#124;</code> | <code>nint operator &#124;(nint left, nint right)</code> | `or` |
+| <code>&#124;</code> | <code>nuint operator &#124;(nuint left, nuint right)</code> | `or` |
+| `^` | `nint operator ^(nint left, nint right)` | `xor` |
+| `^` | `nuint operator ^(nuint left, nuint right)` | `xor` |
+| `<<` | `nint operator <<(nint left, int right)` | `shl` |
+| `<<` | `nuint operator <<(nuint left, int right)` | `shl` |
+| `>>` | `nint operator >>(nint left, int right)` | `shr` |
+| `>>` | `nuint operator >>(nuint left, int right)` | `shr.un` |
+
+For some binary operators, the IL operators support additional operand types
+(see [ECMA-335](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf) III.1.5 Operand type table).
+But the set of operand types supported by C# is limited for simplicity and for consistency with existing operators in the language.
+
+Lifted versions of the operators, where the arguments and return types are `nint?` and `nuint?`, are supported.
+
+Compound assignment operations `x op= y` where `x` or `y` are native ints follow the same rules as with other primitive types with pre-defined operators.
+Specifically the expression is bound as `x = (T)(x op y)` where `T` is the type of `x` and where `x` is only evaluated once.
+
+The shift operators should mask the number of bits to shift appropriately
```

Comment:
>appropriately [](start = 60, length = 13)

You should clarify that "appropriately" means to five bits if `sizeof(nint)` is 4, and to six bits of `sizeof(nint)` is 5. We cannot get that from the quoted section of the spec.